src/payment-providers/tsys/tsys.ts

```diff
@@ -3,176 +3,150 @@ import {Observable} from 'rxjs/Observable';
+  if(!deviceId){
+    return Observable.throw({ message: 'Device ID not set', data: 'init needs to be called first' });
+  }
+  if(!manifest){
+    return Observable.throw({ message: 'Manifest not set', data: 'init needs to be called first' });
+  }
 
-export function encrypt(cardNumber: string, cvv: string, month: number, year: number){
-  // Map from previous observable to wait for the previous encryption to emit a value
-  return previousEncryptObservable = previousEncryptObservable
-    .catch(() => Observable.of('single item')) // convert errors to next emissions so the next observable waits correctly
-    .mergeMap(() => {
-      return exports._importVendorCode(); // Re-import vendor code to retrieve a new public key from TSYS. Use reference on exports obj so it can be mocked
-    })
-    .mergeMap(() => inputsObservable) // After waiting for previous observables to complete we get the TSYS inputs to fill with card details
-    .mergeMap(inputs => {
+  return exports._fetchTsysData()
+    .mergeMap((tsysData: TsysData) => {
       let monthString = String(month);
       monthString = monthString.length === 1 ? '0' + monthString : monthString;
-
-      // Fill TSYS inputs with the values to be encrypted
-      inputs.cardNumber.value = cardNumber;
-      inputs.cvv.value = cvv;
-      inputs.expiryDate.value = monthString + '/' + String(year);
-
-      // Emit response from TSYS tokenization
-      return new Observable<any>((observer: Observer<any>) => {
-        (<any> window).tsepHandler = (eventType: string, event: any) => {
-          switch(eventType){
-            case 'FieldValidationErrorEvent':
-              observer.error(event);
-              observer.complete();
-              break;
-            case 'TokenEvent':
-              if(event && event.status === 'PASS'){
-                observer.next(event);
-              }else{
-                observer.error(event);
-              }
-              observer.complete();
-              break;
-            case 'ErrorEvent':
-              observer.error(event);
-              observer.complete();
-              break;
-          }
-        };
-
-        // Trigger focusout event so that the TSYS library will attempt encrypting and tokenizing the credit card
-        if((<any> window).jqtsep) {
-          (<any> window).jqtsep(inputs.cardNumber).trigger('focusout');
-        }else {
-          (<any> window).$(inputs.cardNumber).trigger('focusout');
-        }
-      });
+      const expiryDate = monthString + '/' + String(year);
+
+      const encryptor = new JSEncrypt();
+      encryptor.setKey(tsysData.key);
+      const encryptedNumber = encryptor.encrypt(cardNumber);
+      if (encryptedNumber === false) {
+        return Observable.throw({ message: 'Encryption error', data: 'Could not encrypt the card number with the key provided by TSYS' });
+      }
+
+      return exports._makeRequest(
+        tsysData.url + '/generateTsepToken',
```

Comment:
Added in https://github.com/CruPCI/cru-payments/pull/13/commits/9c7f9409d2baa5d76e904b94fd0a820313a9e329